### ISSUE TITLE
Makefile replacement

### ISSUE THREAD

**Issue by abitrolly** [bitfield/script#198]
Is it possible to add some primitives to https://github.com/bitfield/script for primitive `Makefile` replacement?

I have no idea how it would look like in Go syntax. Maybe borrow things from SCons? https://scons-cookbook.readthedocs.io/en/latest/

**Comment by bitfield** [comment]
Nice suggestion, @abitrolly, thanks! Maybe you could come up with a few examples of the sort of `script` program you'd like to write to replace a Makefile?

**Comment by abitrolly** [comment]
Well, there is a master issue here https://gitlab.com/gitlab-org/gitlab-development-kit/-/issues/166 which is a general rant about `Makefile` shenanigans.

Maybe something like this would be good for dependency tracking.
```
docs = target('./gitlab-docs).Clone('https://gitlab.com/gitlab-org/gitlab-docs')

main = target.Add(docs)

main.Build()
```

**Comment by msaher** [comment]
Maybe we can start by making `script` a [just](https://github.com/casey/just) replacement first? That covers most of my usecase for `Make`.

**Comment by bitfield** [comment]
@msaher does `just` need replacing?

**Comment by msaher** [comment]
@bitfield, In my opinion? No. I don't dislike writing just files or make files enough to want to replace them with `script`. My original comment was just pointing out that `.PHONY` recipes in `make` already cover most of my use cases. So if script wants to replace `make`, I think it should first start by providing an API for `.PHONY` targets

**Comment by bitfield** [comment]
Yeah, I think `just` is probably the way to go if you need `make`-style functionality (which most people don't)—I can't see what `script` would add here.
